hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/ConsumerMessageSender.java

```diff
@@ -201,21 +208,25 @@ private boolean willExceedTtl(Message message, long delay) {
         return message.isTtlExceeded(remainingTtl);
     }
 
-    private void handleFailedSending(Message message, MessageSendingResult result) {
-        retrySending(message, result);
+    private void handleFailedSending(Message message, MessageSendingResult result, ConsumerProfiler profiler) {
         errorHandlers.forEach(h -> h.handleFailed(message, subscription, result));
+        retrySendingOrDiscard(message, result, profiler);
     }
 
-    private void retrySending(Message message, MessageSendingResult result) {
+    private void retrySendingOrDiscard(Message message, MessageSendingResult result, ConsumerProfiler profiler) {
         List<URI> succeededUris = result.getSucceededUris(ConsumerMessageSender.this::messageSentSucceeded);
         message.incrementRetryCounter(succeededUris);
 
         long retryDelay = extractRetryDelay(message, result);
         if (shouldAttemptResending(message, result, retryDelay)) {
             retries.increment();
-            retrySingleThreadExecutor.schedule(() -> resend(message, result), retryDelay, TimeUnit.MILLISECONDS);
+            profiler.flushMeasurements(ConsumerRun.RETRIED);
+            ConsumerProfiler resendProfiler = subscription.isProfilingEnabled() ? new DefaultConsumerProfiler(subscription.getQualifiedName()) : new NoOpConsumerProfiler();
```

Comment:
I am creating a new one, to have faster feedback, otherwise i would have to wait e.g. 1h for message discarding. I was thinking about adding messageId to the logs, but I am not sure if it is needed right now. So my proposition is to leave it without messageId for now, and we would add this in the future if we need to.